packages/host/app/components/operator-mode/card-schema-editor.gts

```diff
@@ -0,0 +1,163 @@
+      }
+
+      .realm-icon {
+        margin-right: var(--boxel-sp-xxxs);
+      }
+
+      .card-field {
+        background-color: white;
+      }
+
+      .card-field {
+        margin-bottom: var(--boxel-sp-xs);
+        padding: var(--boxel-sp);
+        border-radius: var(--boxel-border-radius);
+        display: flex;
+      }
+
+      .card-fields {
+        margin-top: var(--boxel-sp);
+      }
+
+      .left {
+        display: flex;
+        margin-top: auto;
+        margin-bottom: auto;
+        flex-direction: column;
+      }
+
+      .right {
+        margin-left: auto;
+        margin-top: auto;
+        margin-bottom: auto;
+      }
+
+      .field-name {
+        font-size: var(--boxel-font-size);
+      }
+
+      .field-type {
+        color: #949494;
+      }
+    </style>
+
+    <div
+      class='schema-editor-container'
+      data-test-card-schema={{@cardType.displayName}}
+    >
+      <div class='pill'>
+        <span class='realm-icon'>
+          🟦
+        </span>
+        {{@cardType.displayName}}
+      </div>
+
+      <div class='card-fields'>
+        {{#each @cardType.fields as |field|}}
+          {{#if (this.isOwnField field.name)}}
+            <div class='card-field' data-test-field-name={{field.name}}>
+              <div class='left'>
+                <div class='field-name'>
+                  {{field.name}}
+                </div>
+                <div class='field-type'>
+                  {{field.type}}
+                </div>
+              </div>
+              <div class='right'>
+
+                <div class='pill'>
+                  <span class='realm-icon'>
+                    🟪
+                  </span>
+                  {{#let
+                    (capitalize (this.cleanupCardType (cardId field.card)))
+                    as |cardType|
+                  }}
+                    <span data-test-card-type={{cardType}}>{{cardType}}</span>
+                  {{/let}}
+                </div>
+              </div>
+            </div>
+          {{/if}}
+        {{/each}}
+      </div>
+    </div>
+  </template>
+
+  @service declare loaderService: LoaderService;
+  @service declare cardService: CardService;
+
+  cleanupCardType(value: string) {
+    let path = new URL(value).pathname; // Examples of pathname: "/base/string/default", "/drafts/pet/Pet"
+
+    if (path.endsWith('/default')) {
+      path = path.slice(0, -'/default'.length);
+    }
+
+    let cardType = path.split('/').pop();
+    if (!cardType) {
+      throw new Error(`Could not parse card type from ${value}`);
+    } else {
+      return cardType;
+    }
+  }
+
+  @action
+  isOwnField(fieldName: string): boolean {
+    return Object.keys(
+      Object.getOwnPropertyDescriptors(this.args.card.prototype),
+    ).includes(fieldName);
+  }
+}
+
+function cardId(card: Type | CodeRef): string {
+  if (isCodeRef(card)) {
+    return internalKeyFor(card, undefined);
+  } else {
+    return card.id;
+  }
+}
```

Comment:
let's get rid of this if possible--it was a silly thing we did in the code route that is probably very confusing. card def's should not have "ID's"

Reply:
I adjusted this to not use ids anymore but display names instead. https://github.com/cardstack/boxel/pull/657/commits/50da963f753f74d32f0e47eaa6272327f20732a0